Adds banned characters in name function

ritster opened the pull request:
Disallows you from putting the characters @, *, `, ', and > and " in
your name. Added test cases similar to the MAX_NAME_LENGTH check.

Copied initial code from:
https://github.com/zulip/zulip/pull/2473

smarx commented:
*Automated message from Dropbox CLA bot*

@ritster, thanks for the pull request! It looks like you haven't yet signed the Dropbox CLA. Please [sign it here](https://opensource.dropbox.com/cla/).

smarx commented:
*Automated message from Dropbox CLA bot*
				
@ritster, thanks for signing the CLA!

showell commented on zerver/models.py:
```diff
@@ -483,6 +483,7 @@ class UserProfile(ModelReprMixin, AbstractBaseUser, PermissionsMixin):
 
     USERNAME_FIELD = 'email'
     MAX_NAME_LENGTH = 100
+    NAME_INVALID_CHARS = ['*', '`', '\'', '>', '\"', '@']
```
Just a cosmetic thing, but you can avoid the backslashes by saying `"'"` and `'"'`.

ritster replied:
Ok, I'll fix it. Thanks for the feedback.

showell commented on zerver/tests/tests.py:
```diff
@@ -1615,6 +1623,30 @@ def test_illegal_name_changes(self):
+        user = get_user_profile_by_email(email)
+        full_name = user.full_name
+
+        with self.settings(NAME_CHANGES_DISABLED=True):
```
Was this section copies from somewhere else?

ritster replied:
Yes, this is a modified version of the test: test_illegal_name_changes.

showell replied:
Ok, I don't think you need to duplicate the things that are already tested in the other test.

showell commented:
Thanks for cleaning up the tests, @ritster!  Can you squash your commits down to one commit that's ready to merge?  I think the Travis "coveralls" failure is related to your incremental commit.

https://zulip.readthedocs.io/en/latest/version-control.html#commit-messages

showell commented:
We need to allow single quotes (`'`) in names, for names like O'Hara.

ritster commented:
Ok, @showell. I have squashed the three commits into one and I have changed the code to not disallow  the character ( ' ). Thanks again for the great feedback and advice.

timabbott commented:
Merged, thanks @ritster!  I added a couple commits on top to remove the code duplication and cover a few more code paths; see 2283b5fc916dbf26d98ba6b3e44e802921ecbdff and 56cecc48911445f91b6a5ec95f9d7d67220ee228.  Thanks!!
